### PR TITLE
source google ads: improve custom query interface for full refresh

### PR DESCRIPTION
**Pull request by jpmmcneill**
## What
A new connector image needs to be present to iterate on google ads custom queries that are full refresh. This change keeps that functionality, but improves UX in the feedback loop by making it definable at the custom query level.

## How
Adds a new optional key to each custom query json, that configures whether it represents a full refresh only resource.

## Pre-merge Checklist

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version is set to `0.0.1`
    - [ ] `Dockerfile` has version `0.0.1`
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### CODE REVIEW

**Comment by sh4sh** [comment]
@jpmmcneill thanks for your contribution!

Could you bump `dockerImageTag` in the [connector's metadata.yaml](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-google-ads/metadata.yaml), and update the [connector changelog + docs](https://github.com/airbytehq/airbyte/blob/master/docs/integrations/sources/google-ads.md) with information on how to use the feature?

**Comment by marcosmarxm** [comment]
@sh4sh can you add unit test to ensure the change is safe?

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py [hunk 1]
```diff
@@ -156,12 +156,12 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
             query = single_query_config["query"]
             if self.is_metrics_in_custom_query(query):
                 if non_manager_accounts:
-                    if query.resource_name in FULL_REFRESH_CUSTOM_TABLE:
+                    if query.resource_name in FULL_REFRESH_CUSTOM_TABLE or single_query_config.get("full_refresh_only"):
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:159]: Need to add unit tests, bump version and update change log.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py (hunk 1 above)
Comment (sh4sh) [reply airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:159]: @lazebnyi can you help me understand what additional tests would be valuable? we have a unit test for checking that `is_metrics_in_custom_query` returns a valid response [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-google-ads/unit_tests/test_source.py#L153-L165)

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Thanks for the contribution @jpmmcneill but this is not doing anything. Right now the `single_query_config` object has the `query` and `table_name` there isn't any key called `full_refresh_only`. Can you explain how do you solve or how this solve the problem?

**Comment by marcosmarxm** [comment]
> 

Can you first open a Github Issue to discuss the problem and find the optimal solution?

**Comment by marcosmarxm** [comment]
Closing this contribution because right now it isn't doing the changes proposed in the description.

**Pull request closed without merging**